app/src/main/java/de/danoeh/antennapod/fragment/SubscriptionFragment.java

```diff
@@ -128,11 +166,97 @@ public boolean onOptionsItemSelected(MenuItem item) {
             case R.id.subscription_num_columns_5:
                 setColumnNumber(5);
                 return true;
+            case R.id.subscription_filter:
+                openSubscriptionFilter();
+                return true;
             default:
                 return false;
         }
     }
 
+    private void openSubscriptionFilter() {
+        SubscriptionFilterDialog filterDialog = new SubscriptionFilterDialog(getContext(),
+                autoDownload,keepUpdated,autoDelete) {
+            @Override
+            protected void updateFilter(ArrayList<SubscriptionFilter> filterValues) {
+
+                // do some filtering work
+                // step 1: remove all existing data from current feeds
+                // Step 2: Add all where the filter value is true and assign to currentFeeds
+                currentFeeds = new ArrayList<>();
+                for(SubscriptionFilter filter: filterValues){
+
+                    switch (filter.getOption()) {
+                        case STRING_AUTO_DOWNLOAD:
+                            autoDownload = filter.getValue();
+                            break;
+                        case STRING_KEEP_UPDATED:
+                            keepUpdated = filter.getValue();
+                            break;
+                        case STRING_AUTO_DELETE:
+                            autoDelete = filter.getValue();
+                            break;
+                    }
+                }
+                // edit preference value.
+                prefs.edit().putString(PREF_AUTO_DOWNLOAD,autoDownload).apply();
+                prefs.edit().putString(PREF_KEEP_UPDATED,keepUpdated).apply();
+                prefs.edit().putString(PREF_AUTO_DELETE,autoDelete).apply();
+
+                loadFilter();
+
+            }
+
+        };
+
+        filterDialog.openDialog();
+    }
+
+    private void loadFilter() {
```

Comment:
I think this could be in another file. Something like `currentFeeds = SubscriptionFilter.filter(currentFeeds, autoDownload, keepUpdated, ...)`. The parameters could be enums for each option with On/Off/All. Those enums could also be located in the new class `SubscriptionFilter`

Reply:
okay, I am creating a class. can you suggest a good place where to put class.

Reply:
Maybe in `core/src/main/java/de/danoeh/antennapod/core/feed` next to `FeedItemFilter.java` but if you have a better idea, just use it :) I am not sure what location fits best.